presto-spi/src/test/java/io/prestosql/spi/testing/InterfaceTestUtils.java

```diff
@@ -64,11 +64,23 @@ private InterfaceTestUtils() {}
                     }));
 
             try {
+                if (!defines(forwardingInstance, actualMethod)) {
+                    continue;
```

Comment:
What is this needed for?

Reply:
When forwarding class does not implement all methods.

Reply:
i removed the check and apparently all tests passed
why a forwarding class would want to forward part of the methods only?

isn't the point of a forwarding class to be complete, including interface's `default` methods?

Reply:
> i removed the check and apparently all tests passed

That is correct.

> why a forwarding class would want to forward part of the methods only?

I don't know. That depends on context of forwarding class. Notice that we check if a forwarding method call in separate assertion to assertion that check if all methods are implemented. 
Hence I find these two properties independent.
Typically Forwarding class forward all methods, but it might be not always the case.

Reply:
> Typically Forwarding class forward all methods, but it might be not always the case.

then it's not a true forwarding class. it's dangerous  beast, especially when combined with default interface methods than we use often, 
as you may end up calling eg old deprecated overload even if the target impls the new overload

let me remove this `if`

Reply:
But you have the other assertion to check if all methods are implemented. Running two assertions make sure that you have true forwarding class. I don't like you the fact this assertion might depend on  passing other assertion.

Reply:
https://github.com/prestosql/presto/pull/2618